lib/src/sql/function.rs

```diff
@@ -186,6 +187,34 @@ fn script(i: &str) -> IResult<&str, Function> {
 	Ok((i, Function::Script(v, a)))
 }
 
+fn js_script(i: &str) -> IResult<&str, Function> {
+	/* Design decisions:
+			You can choice if creates a new tag (  E.g fn::js::script()[ ... ]  ) 
+			Or makes compatible the last tags but using a "[" "]" for give users a alternative 
+			when they javascript have conflicts. 
+
+	*/ 
+	//let (i, _) = tag("fn::js::script")(i)?;
+	let (i, _) = alt((tag("fn::script"), tag("fn"), tag("function")))(i)?;
+	let (i, _) = mightbespace(i)?;
+	let (i, _) = tag("(")(i)?;
+	let (i, a) = separated_list0(commas, value)(i)?;
+	let (i, _) = tag(")")(i)?;
+	let (i, _) = mightbespace(i)?;
+	let (i, v) = secure_func(i)?;
+	Ok((i, Function::Script(v, a)))
+}
+
+fn secure_func(i: &str) -> IResult<&str, Script> {
+	let (i, _) = char('[')(i)?;
+	let (i, v) = take_until1("]")(i)?;
+	let (i, _) = char(']')(i)?;
+
+	let secure_js = format!(" try {{{}}} catch (error) {{ return \"SecureCatch::\" + error.message; }}", v);
```

Comment:
What if someone were to input 

```js
}; throw "Wont be caught"; try {
```

?

Reply:
If we study the not-legitimate users activities: **Of course, you are right! my code havent sens for them.**  but for legitimate user my suggestion (i.e. Surreal DB will be able to catch some exceptions) would be useful. But of course, its only a idea...  and the implementation is not good I know it. I attach some diagrams about my suggestions.
https://github.com/surrealdb/surrealdb/pull/1498#issuecomment-1332867874

Congratulations for your project!